src/tribol/interface/tribol.cpp

```diff
@@ -656,14 +719,14 @@ void registerMortarGaps( integer meshId,
 {
    MeshManager & meshManager = MeshManager::getInstance();
 
-   SLIC_ERROR_IF(!meshManager.hasMesh(meshId), "tribol::registerMortarGaps(): " << 
-                 "no mesh with id " << meshId << " exists.");
+   SLIC_ERROR_ROOT_IF(!meshManager.hasMesh(meshId), "tribol::registerMortarGaps(): " << 
+                      "no mesh with id " << meshId << " exists.");
 
    MeshData & mesh = meshManager.GetMeshInstance( meshId );
 
    if (gaps == nullptr)
    {
-      SLIC_WARNING( "tribol::registerMortarGaps(): null pointer to data " << 
+      SLIC_WARNING( "tribol::registerMortarGaps(): null pointer to gap data " << 
```

Comment:
I changed to debug. This is a good point. If the host-code doesn't guard calls for null data, then this call will be made with a nullptr, which I think is ok and will happen as you stated. In a future PR we will make sure we uniformly handle "null-calls" and associated logging.

Reply:
As a note, I do set the mesh to invalid for a call to this routine with a null pointer to the gap data. In principle I could check the given mesh to see if it is a null mesh, but alternatively, it could be the host-codes responsibility to only register non-null data. Handling this more gracefully is a future PR and I'm interested in your input from the implicit testing/side of things.

Reply:
Last thing. I actually changed the if check to only set the mesh to invalid for null pointer with non-null mesh.

Reply:
I don't see the change to SLIC_DEBUG here; is does that still need to be pushed?

Reply:
Never mind, I see the changed if statement!

Reply:
This got a little more confusing. I added checks for null pointers and null vs. non-null meshes. If null pointer and null mesh, it is DEBUG, if null pointer and non-null mesh it is warning and handled later in the code. Again, there is no uniformity to how register calls are handled with null meshes. That will be a future pr.

Reply:
done.